swift-tools-support-core/Sources/TSCUtility/Platform.swift

```diff
@@ -26,8 +27,15 @@ public enum Platform: Equatable {
     /// Lazily checked current platform.
     public static var currentPlatform = Platform._findCurrentPlatform(localFileSystem)
     /// Attempt to match `uname` with recognized platforms.
+    /// Additionally, try `ver` to recognize Windows.
     public static func _findCurrentPlatform(_ fs: FileSystem) -> Platform? {
-        guard let uname = try? Process.checkNonZeroExit(args: "uname").spm_chomp().lowercased() else { return nil }
+        guard let uname = try? Process.checkNonZeroExit(args: "uname").spm_chomp().lowercased() else {
```

Comment:
Could this just use `#if os(Windows)`?  I think that's what @gmittert was doing here in this PR: https://github.com/apple/swift-package-manager/pull/2362 and that seems reasonable to me, since this whole function is about checking the host platform.

Reply:
There does seem to be overlap with that PR in general, but I would defer to @compnerd about how they should be reconciled, since I'm not familiar with Windows.

Reply:
BTW, `#if os(Darwin)` could be done for the `.darwin` case too, though Linux would still need to do a bit more specialized checking.

Reply:
I’ve read #2362, it is adding `.windows` in both SPM and TSC, but the former part was already done so far. Maybe we should get @gmittert here to see whether we could combine the two PRs.

Reply:
Btw, what about the cost of `#if os()` and `Process.checkNonZeroExit()`? I intend to use the more efficient one here because both plans have a high readability and won’t be confusing.

Reply:
I can rebase #2362, however, I don't currently have access to a windows machine for building swift (thanks covid), so if you have an environment set up to test against, it might be easier for you to merge the two and then close #2362 in favour of yours since #2362 is out of date anyways.

`#if os` will be cheaper since it's known at compile time and won't have the cost of spawning a process when checking so that's my recommendation.

Reply:
@compnerd pointed out in https://github.com/apple/swift-tools-support-core/pull/123 that `#if os(Darwin)` won't actually work — I must have been thinking about an earlier forum thread (https://forums.swift.org/t/idea-support-if-os-darwin-as-shorthand-for-os-ios-os-osx-os-watchos-os-tvos/1493) when suggesting it, but it's not actually available.  Apologies for the misdirection.  So for `.darwin`, it might still be necessary to use `uname`, if `#if canImport(Darwin)` isn't dependable as a substitute.

Reply:
I think it's probably not... We should only use `#if canImport(Darwin)` when we actually need it. I'll switch back to `uname`.

Reply:
But there's another thing we can discuss: can we use `#if os(Android)` to distinguish Android? I think that's okay.

Reply:
I don't think that `#if os(Android)` is bad, but its also not something that is great - because I imagine that building on Android is not going to be very common (I suppose that you could be building on a chromebook, but the cross-compilation is a much more persuasive story imo).